src/main/java/com/google/devtools/build/lib/sandbox/LinuxSandboxedSpawnRunner.java

```diff
@@ -95,11 +110,43 @@ public static boolean isSupported(final CommandEnvironment cmdEnv) throws Interr
     return isSupported;
   }
 
-  private static boolean computeIsSupported(CommandEnvironment cmdEnv, Path linuxSandbox)
+  private static SupportLevel computeIsSupported(CommandEnvironment cmdEnv, Path linuxSandbox)
       throws InterruptedException {
+    try {
+      runBinTrue(cmdEnv, linuxSandbox, /* wrapInBusybox= */ false);
+      return SupportLevel.SUPPORTED;
+    } catch (CommandException e) {
+      if (e instanceof BadExitStatusException badExitStatusException) {
+        var stderr = new String(badExitStatusException.getResult().getStderr(), ISO_8859_1);
+        // Ubuntu 24.04 allows unprivileged user namespaces but denies any permissions that wouldn't
+        // be granted outside the namespace. The first syscall that makes use of extended
+        // permissions is a mount.
+        if (stderr.lines().anyMatch(line -> line.endsWith(": \"mount\": Permission denied"))) {
+          try {
+            runBinTrue(cmdEnv, linuxSandbox, /* wrapInBusybox= */ true);
+            return SupportLevel.SUPPORTED_VIA_BUSYBOX;
+          } catch (CommandException e2) {
+            cmdEnv
+                .getReporter()
+                .handle(
+                    Event.warn(
+                        "linux-sandbox failed to run as it lacks permission to use an unprivileged user namespace."
```

Comment:
You might want to include: "Try installing busybox to circumvent AppArmor restrictions".

Reply:
I swear I am not trying to derail this change!
linux-sandbox support on Ubuntu gave me so much headaches so all your changes are <3 

~~~~~~
Well isn't `busybox` mitigation around AppArmor itself tied to some Ubuntu version via a specific installed profile? This whole circumvention could go away anyways in other versions;

Maybe the change should either lean in to this or we can harden against it more.
(i.e. first check if whatever the path is exists  "/etc/apparmor.d/bin.busybox" or maybe `/etc/os-release`)

I as a user would probably like to see this in some diagnostic. I commented on the PR top-level but it feels odd that there is AppArmor for busybox which is a complete escape hatch; I would guess that must be going away in subsequent releases otherwise.... what's the point.

Reply:
Don't worry, thanks for raising these questions. 

The busybox workaround may go away in the future and yes, the new restriction is totally pointless until then (and until the other bypasses have been removed). 

My plan is not to keep bypassing valid security restrictions, but exactly to keep Bazel working until 1) the restriction isn't pointless anymore and 2) there's a better story around allowlisting user applications. That's why I think that relying on a workaround is fine for now, without us committing to keep this working without user action indefinitely.